infra/cifuzz/actions/entrypoint.py

```diff
@@ -49,33 +50,42 @@ def main():
   pr_ref = os.environ.get('GITHUB_REF')
   commit_sha = os.environ.get('GITHUB_SHA')
   event = os.environ.get('GITHUB_EVENT_NAME')
-
-  # Get the shared volume directory and create required directorys.
   workspace = os.environ.get('GITHUB_WORKSPACE')
-  if not workspace:
+
+  # Check if failures should be reported.
+  failure_allowed = (os.environ.get('FAILURE_ALLOWED').lower() == 'true')
+  if not failure_allowed:
+    out_dir = os.path.join(workspace, 'out')
+    os.makedirs(out_dir, exist_ok=True)
+    file_handle = open(os.path.join(out_dir, 'testcase'), "a")
+    file_handle.write('No bugs detected.')
+    file_handle.close()
+
+  if not workspace and failure_allowed:
     logging.error('This script needs to be run in the Github action context.')
     return 1
 
   if event == 'push' and not cifuzz.build_fuzzers(
       oss_fuzz_project_name, github_repo_name, workspace,
-      commit_sha=commit_sha):
+      commit_sha=commit_sha) and failure_allowed:
     logging.error('Error building fuzzers for project %s with commit %s.',
                   oss_fuzz_project_name, commit_sha)
     return 1
   if event == 'pull_request' and not cifuzz.build_fuzzers(
-      oss_fuzz_project_name, github_repo_name, workspace, pr_ref=pr_ref):
+      oss_fuzz_project_name, github_repo_name, workspace,
+      pr_ref=pr_ref) and failure_allowed:
     logging.error('Error building fuzzers for project %s with pull request %s.',
                   oss_fuzz_project_name, pr_ref)
     return 1
 
   # Run the specified project's fuzzers from the build.
   run_status, bug_found = cifuzz.run_fuzzers(oss_fuzz_project_name,
```

Comment:
If we are allowing errors and the build fails, does it still make sense to proceed here?

Reply:
Fixed by changing each error codes values rather than individual conditionals.